m4/curl-functions.m4

```diff
@@ -298,15 +292,13 @@ curl_includes_string="\
 #ifdef HAVE_SYS_TYPES_H
 #  include <sys/types.h>
 #endif
-#ifdef HAVE_STRING_H
-#  include <string.h>
-#endif
+#include <string.h>
 #ifdef HAVE_STRINGS_H
 #  include <strings.h>
 #endif
 /* includes end */"
   AC_CHECK_HEADERS(
-    sys/types.h string.h strings.h,
```

Comment:
This seems to break `strerror_r` detection on Solaris, visible as warnings in the autobuilds since yesterday:
https://curl.se/dev/log.cgi?id=20230927074238-1773871#prob1
```
configure: WARNING: cannot determine strerror_r() style: edit lib/curl_config.h manually.
```

Reply:
Is it possible Solaris doesn't have a string.h? Or it something else? Is there a way to pull or view config.log of a failed build?

Reply:
According to this, strerror_r is declared in string.h: https://docs.oracle.com/cd/E86824_01/html/E54766/strerror-r-3c.html